src/content_script.js

```diff
@@ -296,29 +300,60 @@ function getOffsetsAndApplyClass(elemRect, bodyRect, target, htmlBadgeDiv) {
   }
 }
 
+function isVisible(target) {
+  const styleTransform = ( window.getComputedStyle(target, false).getPropertyValue("transform") === "matrix(1, 0, 0, 0, 0, 0)" );
+  const styleHidden = ( window.getComputedStyle(target, false).getPropertyValue("visibility") === "hidden" );
+  const styleDisplayNone = ( window.getComputedStyle(target, false).getPropertyValue("display") === "none" );
+  if (styleTransform || styleHidden || styleDisplayNone){
+    return false;
+  }
+  return true;
+}
+
 function checkVisibilityAndApplyClass(target, htmlBadgeDiv) {
+
+  if ( target === null ) {
+    htmlBadgeDiv.classList.add("fbc-badge-disabled");
+    return false;
+  }
+
   const htmlBadgeDivHasDisabledClass = htmlBadgeDiv.classList.contains("fbc-badge-disabled");
-  const targetIsNull = (target === null);
 
-  if ( targetIsNull && !htmlBadgeDivHasDisabledClass ) {
-    // Element no longer exists and its badge needs to be hidden
+  if (isDisplayNone(target) && !htmlBadgeDivHasDisabledClass) {
     htmlBadgeDiv.classList.add("fbc-badge-disabled");
-    return;
+    return false;
+  }
+
+  const { parentElement } = target;
+  if (parentElement) {
+    if ( !isVisible(parentElement) ) {
+      if (!htmlBadgeDivHasDisabledClass) {
+        htmlBadgeDiv.classList.add("fbc-badge-disabled");
+      }
+      return false;
+    } else {
+      if ( htmlBadgeDivHasDisabledClass ) {
+        htmlBadgeDiv.classList.remove("fbc-badge-disabled");
+      }
+      return true;
+    }
   }
```

Comment:
Lots of nested `if` blocks here. Would be nice to refactor them into descriptive helper functions if we can.